### DISCUSSION TITLE
Livewire Navigate needs an option to replace state

### DISCUSSION THREAD

**Discussion by starrover** [livewire/livewire#9111] [3. Feature ideas]
My goal is to completely remove the form page from the browser's history, as window.location.replace() does, when redirects to a success page after a form submission. The redirection should utilize navigate:true like $this->redirect('/success', navigate:true). However, current bahavior of Livewire Navigate is pushing state, not replacing, so I have to work around and fire an event to replace the current url to '/success' to be handled by Javascript.  If Navigate cannot replace state,  then Livewire's Navigate feature (which uses pushState) will not achieve this directly. Livewire currently lacks a built-in option to replace the history state instead of pushing it during navigation.

This limitation can indeed diminish the value of using Livewire's Navigate feature in cases where you need to remove the previous page from the history stack. If you are required to use window.location.replace(), you are effectively bypassing Livewire's SPA-like navigation and reverting to full-page reloads. I think a feature like 'navigate: true, replace: true' would be valuable.

**Comment by sparklem-agency** [comment]
waiting for this
